src/test/scala/org/broadinstitute/dsde/agora/server/webservice/EntityCreationPermissionSpec.scala

```diff
@@ -0,0 +1,189 @@
+  }
+
+  override def afterAll() = {
+    clearDatabases()
+  }
+
+  behavior of "Agora, when creating/editing entities (methods and configs)"
+
+  it should "set up test fixtures correctly" in {
+    val ns1perms = patiently(permissionBusiness.listNamespacePermissions(AgoraEntity(testEntity1.namespace), owner1.get))
+
+    val expectedNs1Perms = List(AccessControl(owner1.get, AgoraPermissions(All)))
+
+    assertResult(expectedNs1Perms) {ns1perms}
+
+    val ns2perms = patiently(permissionBusiness.listNamespacePermissions(AgoraEntity(testEntity2.namespace), owner2.get))
+
+    val expectedNs2Perms = List(
+      AccessControl(owner2.get, AgoraPermissions(All)),
+      AccessControl(owner3.get, AgoraPermissions(All))
+    )
+
+    assertResult(expectedNs2Perms) {ns2perms}
+  }
+
+  it should "allow creating a new entity if you have permissions on the namespace" in {
+    // owner1 create in namespace1
+    assertEntityCreation(owner1.get, namespace1.get, randUUID)
+  }
+
+  it should "reject creating a new entity if you lack permissions on the namespace" in {
+    // owner1 create in namespace2
+    assertEntityRejection(owner1.get, namespace2.get, randUUID)
+  }
+
+  it should "allow creating a new snapshot of an existing entity if you own the entity" in {
+    // owner1 create new snapshot of testEntity1WithId in namespace1
+    assertEntityCreation(owner1.get, testEntity1WithId.namespace.get, testEntity1WithId.name.get)
+  }
+
+  it should "reject creating a new snapshot of an existing entity if you do not own the entity" in {
+    // owner3 create random new entity in namespace2: prove that owner3 has Create in the namespace
+    assertEntityCreation(owner3.get, namespace2.get, randUUID)
+
+    // assert that owner3 does not have any permissions on pre-existing testEntity2WithId
+    val existingPerms = patiently(permissionBusiness.listEntityPermissions(testEntity2WithId, owner2.get))
+    assert(!existingPerms.exists(_.user == owner3.get), "owner3 should not have any permissions on testEntity2WithId")
+
+    // owner3 create new snapshot of testEntity2WithId in namespace2: should fail to create, because owner3 doesn't
+    // have permissions on previous snapshots.
+    assertEntityRejection(owner3.get, testEntity2WithId.namespace.get, testEntity2WithId.name.get)
+  }
+
+  it should "allow creating a new snapshot of an existing entity if you own a snapshot of the entity but lack permission on the namespace" in {
+    val name = randUUID
+    assertEntityCreation(owner1.get, namespace1.get, name)
+
+    val ent = AgoraEntity(namespace1, Some(name), Some(1))
+
+    // add owner2
+    patiently(permissionBusiness.insertEntityPermission(
+      ent, owner1.get, AccessControl(owner2.get, AgoraPermissions(All))
+    ))
+    // remove owner1
+    patiently(permissionBusiness.deleteEntityPermission(
```

Comment:
Can you explain why it's necessary to remove permission for `owner1`? It seems like the test only cares about what `owner2` can do